src/components/Table/utils/columns.tsx

```diff
@@ -169,14 +169,17 @@ export function assignDefaultColumnIds<T extends Kinded>(columns: GridColumn<T>[
   // exists as part of `selectColumn` and `collapseColumn`.
   return columns.map((c, idx) => {
     const { expandColumns } = c;
-    const expandColumnsWithId: GridColumnWithId<T>[] | undefined = expandColumns?.map((ec, ecIdx) => ({
-      ...ec,
-      id: ec.id ?? (`${generateColumnId(idx)}_${ecIdx}` as string),
-      // Defining this as undefined to make TS happy for now.
-      // If we do not explicitly set to `undefined`, TS thinks `expandColumns` could still be of type GridColumn<T> (not WithId).
-      // We only support a single level of expanding columns, so this is safe to do.
-      expandColumns: undefined,
-    }));
+
+    const expandColumnsWithId = isFunction(expandColumns)
```

Comment:
Suggestion: Could we use `getExpandedColumns(c)` here? Just to reuse some of the isFunction/etc. handling.

Reply:
yes, I see the case for reusability, but I think there are 2 issues: the `column` types aren't the same - `getExpandedColumns` wants a `GridColumnWithId<any>` and `assignDefaultColumnIds` takes in a column with type `GridColumn<T>`. 

I also need access to `tableState` to call `getExpandedColumns` and since this is just a regular function, I can't use the `TableStateContext` to get `tableState`

Reply:
Cool, sg!